tests/Unit/Framework/Tests/Test_ActionTesting.cpp

```diff
@@ -1171,21 +1172,19 @@ SPECTRE_TEST_CASE("Unit.ActionTesting.MutableGlobalCache", "[Unit]") {
 
   auto& cache = ActionTesting::cache<component>(runner, 0);
   auto& element_proxy = ::Parallel::get_parallel_component<component>(cache)[0];
+  const auto array_component_id =
+      Parallel::make_array_component_id<component>(static_cast<int>(0));
 
   CHECK(Parallel::mutable_cache_item_is_ready<CacheTag>(
-      cache, [](const int /*value*/) {
+      cache, array_component_id, [](const int /*value*/) {
         return std::unique_ptr<Parallel::Callback>{};
       }));
 
+  // We cannot have two different types of callbacks at once in the mutable
```

Comment:
Should we add a `\warning` somewhere to the documentation that states this clearly and gives a brief summary of what's needed to generalize?